src/ngraph/dimension.cpp

```diff
@@ -103,3 +103,32 @@ bool Dimension::merge(Dimension& dst, const Dimension d1, const Dimension d2)
         return true;
     }
 }
+
+bool Dimension::merge_bcast(Dimension& dst, const Dimension d1, const Dimension d2)
```

Comment:
Minor nit: I'd prefer to spell out `broadcast`.

Reply:
Not sure if there is anything numpy-specific in this behavior but we might think about it and name the function accordingly if so. (Can't imagine what other behaviors would make sense at the dimension level, though.)

Reply:
Might rename to `broadcast_merge` for consistency with `merge_into_broadcast` below. (I dunno, maybe it's technically not inconsistent since one has the preposition.)

Reply:
Can you make this return a result instead of via an arg? That or make `this` serve as `dst` rather than a static.  I think the static methods are leftovers from a different implementation of `Dimension`.

Reply:
@diyessi The statics are still in use. `merge` (like the new `merge_bcast`) can fail, and callers may want to recover gracefully from the failure, so I had made `merge` return `bool`. (Generally we are not using try/catch for situations like this in nGraph.)

Reply:
i dont think there is any numpy-specific behavior here. Most of that happens at the Shape level where there seem to be different approaches on how to line up axes.